solutions/usaco-592.mdx

```diff
@@ -69,4 +72,62 @@ int main() {
 ```
 
 </CPPSection>
+<JavaSection>
+
+```java
+import java.util.*;
+import java.io.*;
+
+public class Angry {
+	static int N, A = 0;
+	static int[] positions;
+	public static void main(String[] args) throws Exception {
+		Kattio io = new Kattio("angry");
+
+		N = io.nextInt();
+		positions = new int[N];
+
+		for (int i = 0; i < N; i++) {
+			positions[i] = io.nextInt();
+		}
+
+		Arrays.sort(positions);
+
+		for (int i = 0; i < N; i++) {
+			// Computing answer
+			A = Math.max(A, getNum(i,-1) + getNum(i, 1) + 1);
+		}
+
+		io.println(A);
+		io.close();
+	}
+
+	public static int getNum(int start, int direction) {
+		int radius = 1;
+		int prev = start;
+
+		while (true) {
+			int next = prev;
+			// Getting the furthest explosion index without exceeding radius
+			while (next + direction >= 0 && next + direction < N 
+				&& Math.abs(positions[next + direction] - positions[prev]) <= radius) {
+				next += direction;
+			}
+			// We didn't find a new haybale, so the chain explosion is over
+			if (next == prev) {
+				break;
+			}
+			// Update our previous explosion and increment radius
+			prev = next;
+			radius++;
+		}
+
```

Comment:
ok went over it again there's still some bad newlines
specifically 46, 56, 58, 62, 69 (also why is there no endl or \n there), 89, 93, 100, 108, and 124